index.php

```diff
@@ -567,30 +568,55 @@ private function getUpdateServerResponse(): array {
 	/**
 	 * Downloads the nextcloud folder to $DATADIR/updater-$instanceid/downloads/$filename
 	 *
+	 * Logs download progress
+	 * Resumes incomplete downloads if possible
+	 * Supports outbound proxy usage
+	 * Logs download statistics upon completion
+	 *
+	 * TODO: Provide download progress in real-time (in both CLI and Web modes)
+	 *
 	 * @throws \Exception
 	 */
 	public function downloadUpdate(): void {
 		$this->silentLog('[info] downloadUpdate()');
 
 		$response = $this->getUpdateServerResponse();
-
-		$storageLocation = $this->getUpdateDirectoryLocation() . '/updater-'.$this->getConfigOptionMandatoryString('instanceid') . '/downloads/';
-		if (file_exists($storageLocation)) {
-			$this->silentLog('[info] storage location exists');
-			$this->recursiveDelete($storageLocation);
-		}
-		$state = mkdir($storageLocation, 0750, true);
-		if ($state === false) {
-			throw new \Exception('Could not mkdir storage location');
-		}
-
 		if (!isset($response['url']) || !is_string($response['url'])) {
 			throw new \Exception('Response from update server is missing url');
 		}
 
-		$fp = fopen($storageLocation . basename($response['url']), 'w+');
+		$storageLocation = $this->getUpdateDirectoryLocation() . '/updater-'.$this->getConfigOptionMandatoryString('instanceid') . '/downloads/';
+		$saveLocation = $storageLocation . basename($response['url']);
```

Comment:
```suggestion
		$saveLocation = $storageLocation . basename($response['url']);
		$this->previousProgress = 0;
```
Just to be cleaned I would reset it there to make sure we behave correctly if downloadUpdate is done twice with the same object.

Reply:
Done.